src/linear_algebra/invariant_basis_number.lean

```diff
@@ -96,6 +97,24 @@ begin
     ((P.injective.comp i).comp Q.injective)
 end
 
+/-- A ring satisfies the strong rank condition if and only if, for all `n : ℕ`, there are no
+injective linear maps `(fin (n + 1) → R) →ₗ[R] (fin n → R)`. -/
+lemma strong_rank_condition_iff : strong_rank_condition R ↔
+  ∀ (n : ℕ) (f : (fin (n + 1) → R) →ₗ[R] (fin n → R)), ¬function.injective f :=
+begin
+  refine ⟨λ h n, _, λ h, ⟨λ n m f hf, _⟩⟩,
+  { rintros ⟨f, hf⟩,
+    letI : strong_rank_condition R := h,
+    exact nat.not_succ_le_self n (le_of_fin_injective R f hf) },
+  { by_contra H,
+    obtain ⟨k, hk⟩ := nat.le.dest (nat.succ_le_iff.mpr (not_le.1 H)),
+    exact h m ⟨f.comp ((linear_equiv.Pi_congr_left' R (λ (i' : fin (m + 1 + k)), R)
+      (fin_congr hk)).to_linear_map.comp (ext_zero_fin R (m + 1) k)), function.injective.comp hf
+      (function.injective.comp (function.injective.comp ((equiv.Pi_congr_left'
+      (λ (i' : fin (m + 1 + k)), R) (fin_congr hk)).injective) (ext_zero_fin.injective R (m + 1) k))
+      function.injective_id)⟩ }
+end
```

Comment:
Fixed:
```suggestion
begin
  refine ⟨λ h n, _, λ h, ⟨λ n m f hf, _⟩⟩,
  { rintros f hf,
    letI : strong_rank_condition R := h,
    exact nat.not_succ_le_self n (le_of_fin_injective R f hf) },
  { by_contra H,
    obtain ⟨k, hk⟩ := nat.le.dest (nat.succ_le_iff.mpr (not_le.1 H)),
    exact h m
      (f.comp ((linear_equiv.Pi_congr_left' R (λ (i' : fin (m + 1 + k)), R)
        (fin_congr hk)).to_linear_map.comp (ext_zero_fin R (m + 1) k)))
      (hf.comp $
        (equiv.Pi_congr_left' (λ (i' : fin (m + 1 + k)), R) (fin_congr hk)).injective.comp $
          ext_zero_fin.injective R (m + 1) k) }
end
```

Reply:
Thank you! If you have more interesting stuff to do don't waste your time fixing failure in my PR, I am happy to do it (I was interrupted by some teaching stuff...) :)